readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -23,48 +23,48 @@ function attach_elastic_search_query(data) {
 
         search_def
-            .then(function (results) {
-                var hits = results.hits || {};
-                var hit_list = hits.hits || [];
+            .then(function (data) {
+                var hit_list = data.results || [];
+                var total_count = data.count || 0;
 
                 if (hit_list.length) {
-                    for (var n in hit_list) {
-                        var hit = hit_list[n];
-                        var fields = hit.fields || {};
-                        var list_item = $('<li style="display: none;"></li>');
-                        var item_url = document.createElement('a');
-                        var highlight = hit.highlight;
-
-                        item_url.href += fields.link +
-                            DOCUMENTATION_OPTIONS.FILE_SUFFIX;
-                        item_url.search = '?highlight=' + $.urlencode(query);
-
-                        // Result list elements
-                        list_item.append(
-                            $('<a />')
-                            .attr('href', item_url)
-                            .html(fields.title)
-                        );
-                        // fields.project is returned as an array
-                        if (fields.project.indexOf(project) === -1) {
-                            list_item.append(
-                                $('<span>')
-                                .text(" (from project " + fields.project + ")")
-                            );
+                    for (var i = 0; i < hit_list.length; i += 1) {
+                        var document = hit_list[i];
+                        var highlight = document.highlight;
+                        var $list_item = $('<li style="display: none;"></li>');
+
+                        // Creating the result from elements
+                        var link = document.link + DOCUMENTATION_OPTIONS.FILE_SUFFIX +
+                                   '?highlight=' + $.urlencode(query);
+
+                        var $item = $('<a>', {'href': link});
+                        $item.html(document.title);
+                        $list_item.append($item);
+
+                        // If the document is from subproject, add extra information
+                        if (document.project !== project) {
+                            var text = " (from project " + document.project + ")";
+                            var $extra = $('<span>', {'text': text});
+
+                            $list_item.append($extra);
                         }
-                        if (highlight.content.length) {
-                            var content = $('<div class="context">')
-                                .html(xss(highlight.content[0]));
-                            content.find('em').addClass('highlighted');
-                            list_item.append(content);
+
+                        // Show highlighted texts
+                        if (highlight.content) {
```

Comment:
Do we not need to check `length` here? I don't know JS well enough, but that's what we were doing before, so probably worth keeping it explicit.

Reply:
In the previous API, the `content` was provided as a blank list if there are no match in content. but our current API do not provide anything in the `content` key if there are not anything in the content. So checking the length may raise error.